core/src/main/java/com/linecorp/armeria/client/retry/RetryingHttpClient.java

```diff
@@ -169,13 +169,15 @@ private void retry0(int currentAttemptNo, Backoff backoff, ClientRequestContext
                         DefaultHttpResponse rootResponse, Throwable exception, long millisAfter) {
         long nextDelay = backoff.nextDelayMillis(currentAttemptNo);
         if (nextDelay < 0) {
+            rootReqDuplicator.close();
             rootResponse.close(exception);
```

Comment:
I think rootReqDuplicator.close() ahead of rootResponse.close() is better even thought it doesn't matter. The flow is like this: 1. client send HttpRequest 2. HttpRequest is wrapped by Duplicator 3. ChildRequest is sent to the server while rootResponse is returned to the client 4. If the RetryingClient doesn't decide to retry again ChildRequest is already subscribed so don't need the Duplicator 5. So clean the reqDuplicator before returning exception to the client..